### DISCUSSION TITLE
A question about working with Array

### DISCUSSION THREAD

**Discussion by Kolhun** [nlohmann/json#4328] [Q&A] [answered]
Hello dear normann, I have a pre-assembled json that gives a strange error when broken down, what could be the reason? There are no problems with dataResultNew["data"], it unloads as a string. Screenshot with an error in the attachment

![image](https://github.com/nlohmann/json/assets/36815965/fdf9cee4-a7e6-49ad-9e91-ec8e7fd474e2)

```cpp
struct DataItem
{
    std::string ID = "";
    std::string Parent = "";
    std::string ParentEnabled = "";

};

void from_json(const nlohmann::json& j, DataItem& d) {
    j.at("id").get_to(d.ID);
    j.at("parent").get_to(d.Parent);
    j.at("parentEnabled").get_to(d.ParentEnabled);
}

std::vector<std::string> split(const std::string& s, char delimiter) {
    std::vector<std::string> tokens;
    std::string token;
    std::istringstream tokenStream(s);
    while (std::getline(tokenStream, token, delimiter)) {
        tokens.push_back(token);
    }
    return tokens;
}

int main() 
{
///
std::string dataSuperString = dataResultNew["data"].dump();
auto dataSuper = json::parse(dataSuperString.c_str()).get<std::vector<DataItem>>();

for (const auto& d : dataSuper)
{
    if (!d.Parent.empty())
    {
        auto parents = split(d.Parent, ',');
        for (const auto& p : parents) {
            parentToChildren[p].push_back(d.ID);
            childToParent[d.ID] = p;
        }
    }
}

return 0;
}
```

**Comment by gregmarr** [comment, marked as answer]
What's your call stack?  This is telling you that you are trying to read an array, but you're reading something else.  There isn't enough information there to see what it is that you're trying to read.

**Reply by Kolhun** [reply]
> What's your call stack? This is telling you that you are trying to read an array, but you're reading something else. There isn't enough information there to see what it is that you're trying to read.

Swears at this particular line, although John is not empty and everything is OK with the structure

```cpp
auto dataSuper = json::parse(dataSuperString.c_str()).get<std::vector<DataItem>>();
```

![image](https://github.com/nlohmann/json/assets/36815965/dd51058c-ba3c-45aa-b5b2-11358d9992f1)
